### DISCUSSION TITLE
Upload API add missing "populate" parameter

### DISCUSSION THREAD

**Discussion by weidox** [strapi/strapi#7923] [Legacy Topics]
`/node_modules/strapi-plugin-upload/services/Upload.js`
All methods do not have "populate" parameter (where it would be applicable), unlike regular API services. For example:
```
  search(params) {
    return strapi.query('file', 'upload').search(params);
  }, 
```
This one should be written as:
```
  search(params, populate) {
    return strapi.query('file', 'upload').search(params, populate);
  }, 
```

This means that service API is not very much usable, because to get populate parameter it forces to use `strapi.query()` API.

**Comment by lauriejim** [comment]
Hello! 
Can you please update your comment to have a readable format.
It' quite not readable.
Thank you this will help us to help you.

**Reply by weidox** [reply]
Done, if that would be enough.

**Comment by derrickmehaffy** [comment]
The search query does not have a population option, can you please open a feature request github issue instead?
